rust-runtime/aws-smithy-http-server/src/proto/aws_json/rejection.rs

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+use strum_macros::Display;
+
+use crate::rejection::MissingContentTypeReason;
+
+#[derive(Debug, Display)]
+pub enum ResponseRejection {
+    InvalidHttpStatusCode,
+    Serialization(crate::Error),
```

Comment:
Given that we're now protocol specific presumably we don't need to use `crate::Error` (~ `Box<dyn Error>`)? 

A task for another PR?

Reply:
Yeah I'm going to remove some of these first.

Reply:
Starting with e.g. https://github.com/awslabs/smithy-rs/pull/2522.